apps/ui/src/composables/usePayment.ts

```diff
@@ -0,0 +1,168 @@
+import { BigNumber } from '@ethersproject/bignumber';
+import { Contract } from '@ethersproject/contracts';
+import { toUtf8Bytes } from '@ethersproject/strings';
+import { abis } from '@/helpers/abis';
+import { verifyNetwork } from '@/helpers/utils';
+import { ChainId } from '@/types';
+
+export type Token = {
+  address: string;
+  decimal: number;
+  symbol: string;
+  chainId: ChainId;
+  logo: string;
+};
+
+// TODO: Double check contract address for each network
+const PAYMENT_CONTRACT_ADDRESSES: Record<ChainId, string> = {
+  1: '',
+  11155111: '0xA92D665c4814c8E1681AaB292BA6d2278D01DEE0',
+  8453: '0xA92D665c4814c8E1681AaB292BA6d2278D01DEE0',
+  84532: ''
+} as const;
+
+const PAYMENT_CONTRACT_ABI = [
+  'function payWithERC20Token(address token, uint256 amount, bytes barcode)'
+] as const;
+
+export const ASSETS = {
+  USDC: {
+    decimal: 6,
+    symbol: 'USDC',
+    logo: 'ipfs://bafybeienhj4ymzglig6at2bovugjhb6rj4xlfloj2rc4lqad57zwi4zs54'
```

Comment:
We should use Stamp token image resolver here instead.

Reply:
Stamp does not return any result for base and all sepolia tokens

Reply:
It works, check here, maybe you miss the network prefix https://snapshot.box/#/base:0x282d013BF31374D0046F04D6a7644d97751ed339/treasury/1/tokens

Reply:
![localhost_8080_](https://github.com/user-attachments/assets/e41ed4ed-5651-4e43-9518-918d6c54c5de)

Was missing the network prefix yes, but logo still missing on testnet, but no big deal